pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -0,0 +1,118 @@
+#include "ndarrays.h"
+
+/* 
+** allocation
+*/
+
+t_ndarray array_init(char *temp, int nd, int *shape, enum e_types type, int type_size)

```

Comment:
Why do we need to pass the type_size? Can `array_init` not define this using the type?